lib/dor/text_extraction/ocr.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Dor
+  module TextExtraction
+    # Determine if OCR is required for a given object
+    class Ocr
+      attr_reader :cocina_object, :workflow_context
+
+      def initialize(params = {})
+        @cocina_object = params[:cocina_object]
+        @workflow_context = params[:workflow_context]
+      end
+
+      def required?
+        # only items can be OCR'd
+        return false unless cocina_object.dro?
+
+        # only specific content types can be OCR'd
+        return false unless allowed_object_types.include? cocina_object.type
+
+        # checks if user has indicated that OCR should be run (set as workflow_context)
+        return false unless workflow_context['runOCR']
+
+        # check to be sure there are actually files to be OCRed
+        return false unless filenames_to_ocr.any?
+
+        # TODO: check for any files that have "manuallyCorrected" in cocina structural (then skip)
+
```

Comment:
raise an error for cases where the user asked for OCR but it is not possible instead of just silently continuing